### PR TITLE
Basic authentication

### PR DESCRIPTION
**Pull request by furkanilhan**
#3

### CODE REVIEW

File: plugins/jwt.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+const fp = require('fastify-plugin')
+
+module.exports = fp(async function (fastify, opts) {
+  fastify.register(require('fastify-jwt'), {
+    secret: 'supersecret'
```
Comment (cemremengu) [inline plugins/jwt.js]: `secret: process.env.SECRET || opts.auth.secret`

**Comment by mcollina** [comment]
I would use a bearer auth instead.

Also, I would specify a hook for the full plugin to check if a route is authorized or not.

**Comment by cemremengu** [comment]
> I would use a bearer auth instead.

@mcollina sorry could you please elaborate on this? Currently its basic auth for credentials exchange for a JWT. After that only token is used for auth. You mean removing the basic auth and only using jwt or there is a better way to send credentials?

**Comment by mcollina** [comment]
I recommend to use https://github.com/fastify/fastify-bearer-auth instead. Basic won't work well in browsers etc.

**Comment by cemremengu** [comment]
> Also, I would specify a hook for the full plugin to check if a route is authorized or not.

@mcollina what do you suggest for skipping authorization for only a single route (i.e. the one that returns the token) since using a global hook will protect all endpoints?

**Comment by mcollina** [comment]
hooks are encapsulated, so you can use a hook for all routes defined within a plugin. That should get the job done.

**Comment by cemremengu** [comment]
@mcollina Removed basic auth and now sending user/pass directly for a token. I refrained from using  bearer auth since that would have been like a static API key type of authentication I think.

Would you mind giving one last look at this when available? 🙏

**Review by mcollina — APPROVED** [review]
LGTM

**Pull request merged**